java/core/src/java/org/apache/orc/Reader.java

```diff
@@ -265,7 +269,7 @@ public Options schema(TypeDescription schema) {
      *
      * @return this
      */
-    public Options setRowFilter(String[] filterColumnNames, Consumer<VectorizedRowBatch> filterCallback) {
+    public Options setRowFilter(String[] filterColumnNames, Consumer<FilterContext> filterCallback) {
```

Comment:
This looks like a breaking API change.

Reply:
Yes this is a breaking change. I thought the convenience of retrieving the column vectors via name does ease the complexity of filter development.

I am open to other ideas on achieving the same.

Reply:
As you know, we cannot change like this. You may try to add as an additional new function instead.

Reply:
This is not yet released in ORC so I was expecting that we should be flexible on changing this.

Reply:
Merged into master as part of ORC-755